### PR TITLE
[TRAFODION-2650] Reorganized Trafodion Configuration code into separate directory structure

### PR DESCRIPTION
**Pull request by zcorrea**
Moved code out of monitor source tree into its own directory structure. Moved Trafodion Configuration API header file into the export/include/trafconf directory.

NOTE: This is an initial step to prepare the code base before adding the MySQL storage method.

### CODE REVIEW

**Comment by zcorrea** [comment]
Jenkins, retest

**Comment by DaveBirdsall** [comment]
Could you explain how these changes resolve the issue raised in JIRA TRAFODION-2650?

File: core/sqf/sqenvcom.sh [hunk 1]
```diff
@@ -47,6 +47,15 @@ export PRODUCT_COPYRIGHT_HEADER="2015-2017 Apache Software Foundation"
 ##############################################################
 export TRAFODION_ENABLE_AUTHENTICATION=${TRAFODION_ENABLE_AUTHENTICATION:-NO}
 
+# Set the Trafodion Configuration store type
+if [[ -n "$CLUSTERNAME" ]]; then
+  # This is a cluster environment, not a workstation
+  #export TRAF_CONFIG_DBSTORE=MySQL
+  export TRAF_CONFIG_DBSTORE=Sqlite
+else
+  export TRAF_CONFIG_DBSTORE=Sqlite
```
Comment (DaveBirdsall) [inline core/sqf/sqenvcom.sh]: Looks like it is aways Sqlite. Why bother with the "if"? Earlier I saw mention in the code that Sqlite is deprecated. Is that still the case?

File: core/sqf/export/include/trafconf/trafconfig.h [hunk 2]
```diff
@@ -67,26 +68,64 @@
 #define PERSIST_RETRIES_KEY        "PERSIST_RETRIES"
 #define PERSIST_ZONES_KEY          "PERSIST_ZONES"
 
-enum TC_STORAGE_TYPE {
+#define TC_ROOT_NODE               "/trafodion"
+#define TC_INSTANCE_NODE           "/instance"
+
+typedef enum {
+    ProcessType_Undefined=0,                // No process type as been defined
+    ProcessType_TSE,                        // Identifies a Table Storage Engine (DP2)
```
[the remark this answers is not in the document]
Comment (zcorrea) [reply core/sqf/export/include/trafconf/trafconfig.h]: No, and to remove them will require additional work in the Seabed layer. Perhaps another JIRA to address this code cleanup here and other areas will be necessary,

File: core/sqf/sqenvcom.sh (hunk 1 above)
Comment (zcorrea) [reply core/sqf/sqenvcom.sh]: This is an initial step to prepare the code base before adding the MySQL storage method.

**Comment by DaveBirdsall** [comment]
Thanks for the responses. I will merge this today.

**Pull request merged**